### PR TITLE
Adding package install/uninstall script

### PR DESCRIPTION
**Pull request by PraveenPenguin**
These script can be used as pre/post avocado mechanism

Signed-off-by: Praveen K Pandey <praveen@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by narasimhan-v** [comment]
@PraveenPenguin Is there any additional config needed ?
I do not see any invocation of pre/post scripts.

```
localhost:~/wrapper_testing/tests # ./avocado-setup.py --bootstrap
15:13:57 INFO    : Check for environment
15:13:57 INFO    : Creating Avocado Config
15:13:59 INFO    : Bootstrapping Avocado
15:13:59 INFO    : Cloning the repo: avocado in /root/wrapper_testing/tests/avocado
15:14:00 INFO    : Installing avocado from /root/wrapper_testing/tests/avocado
15:14:02 INFO    : Cloning the repo: avocado-misc-tests in /root/wrapper_testing/tests/tests/avocado-misc-tests
15:14:03 INFO    : copied all files from  /root/wrapper_testing/tests/config/prescript to /etc/avocado/scripts/job/pre.d/
15:14:03 INFO    : copied all files from  /root/wrapper_testing/tests/config/postscript to /etc/avocado/scripts/job/post.d/
15:14:04 WARNING : Avocado html plugin not installed
15:14:04 INFO    : Installing optional plugin: html
15:14:05 WARNING : Avocado varianter_yaml_to_mux plugin not installed
15:14:05 INFO    : Installing optional plugin: varianter_yaml_to_mux
15:14:05 INFO    : Removing temporary mux dir

localhost:~/wrapper_testing/tests # avocado run /bin/true 
JOB ID     : cd32441f433d7a4957533bf02fd1cd66423ec759
JOB LOG    : /root/wrapper_testing/tests/results/job-2020-05-19T15.14-cd32441/job.log
 (1/1) /bin/true: PASS (0.25 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 0.43 s
JOB HTML   : /root/wrapper_testing/tests/results/job-2020-05-19T15.14-cd32441/results.html
localhost:~/wrapper_testing/tests # vi /root/wrapper_testing/tests/results/job-2020-05-19T15.14-cd32441/job.log
localhost:~/wrapper_testing/tests # 


localhost:~/wrapper_testing/tests # grep script /root/wrapper_testing/tests/results/job-2020-05-19T15.14-cd32441/job.log
localhost:~/wrapper_testing/tests # 
```

**Comment by sathnaga** [comment]
@PraveenPenguin this is not per-Job pre/post plugin usecase, I do think this is relevant for wrapper by default.

File: config/prescript/package_install.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+
```
Comment (narasimhan-v) [inline config/prescript/package_install.sh:3]: extra empty line

File: config/postscript/package_remove.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+package=('pep8' 'flake8' 'nose' 'nose2' 'epydoc' 'pylint' 'pexpect' 'netifaces' 'paramiko' 'enum' 'configparser')
+
+for i in "${package[@]}"; 
+do 
+	echo "un install pip package "$i
+	pip uninstall $i
```
Comment (narasimhan-v) [inline config/postscript/package_remove.sh:8]: `pip uninstall $i -q -y`

File: config/postscript/package_remove.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+package=('pep8' 'flake8' 'nose' 'nose2' 'epydoc' 'pylint' 'pexpect' 'netifaces' 'paramiko' 'enum' 'configparser')
+
+for i in "${package[@]}"; 
+do 
+	echo "un install pip package "$i
```
Comment (narasimhan-v) [inline config/postscript/package_remove.sh:7]: uninstalling

File: config/prescript/package_install.sh [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+package=('pep8' 'flake8' 'nose' 'nose2' 'epydoc' 'pylint' 'pexpect' 'netifaces' 'paramiko' 'enum' 'configparser')
+
+for i in "${package[@]}"; 
+do 
+	echo "install pip package "$i
```
Comment (narasimhan-v) [inline config/prescript/package_install.sh:8]: installing

File: config/prescript/package_install.sh [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+	pip install $i
```
Comment (narasimhan-v) [inline config/prescript/package_install.sh:9]: `pip install $i -q`

**Comment by narasimhan-v** [comment]
@PraveenPenguin @sathnaga How about a quiet output like below:

```
localhost:~ # bash /etc/avocado/scripts/job/pre.d/package_install.sh 
installing package pylint failed
installing package paramiko failed
installing package enum failed
localhost:~ # bash /etc/avocado/scripts/job/post.d/package_remove.sh
localhost:~ # 
```


```
#!/bin/bash

package=('pep8' 'flake8' 'nose' 'nose2' 'epydoc' 'pylint' 'pexpect' 'netifaces' 'paramiko' 'enum' 'configparser')

for i in "${package[@]}"; 
do 
	pip install $i -q > /dev/null 2>&1
	if [[ $? != 0 ]];
	then
	    echo "installing package $i failed"
	fi
done
```
```
#!/bin/bash

package=('pep8' 'flake8' 'nose' 'nose2' 'epydoc' 'pylint' 'pexpect' 'netifaces' 'paramiko' 'enum' 'configparser')

for i in "${package[@]}"; 
do 
	pip uninstall $i -y -q > /dev/null 2>&1
        if [[ $? != 0 ]];
	then
	    echo "uninstall of package $i failed"
        fi    
done
```

**Comment by narasimhan-v** [comment]
This patch works, but needs execute permission for those files. Fixed via https://github.com/open-power-host-os/tests/pull/216

**Comment by sathnaga** [comment]
It is not good to have a custom scripts introduced in wrapper which is set to run default for all jobs even if it does not need.
wrapper provides mechanism to have custom scripts and it is upto users ,how to have them in their env, so will close this PR,
please reopen incase of further discussion on it, tnx!

**Pull request closed without merging**